src/zerocash/PourTransaction.cpp

```diff
@@ -293,65 +289,45 @@ void PourTransaction::init(uint16_t version_num,
-    std::string nonce_new_1_string(nonce_new_1_bytes, nonce_new_1_bytes + ZC_RHO_SIZE);
-    std::string nonce_new_2_string(nonce_new_2_bytes, nonce_new_2_bytes + ZC_RHO_SIZE);
-    std::string rand_new_1_string(rand_new_1_bytes, rand_new_1_bytes + ZC_R_SIZE);
-    std::string rand_new_2_string(rand_new_2_bytes, rand_new_2_bytes + ZC_R_SIZE);
-
-    AutoSeededRandomPool prng_1;
-    AutoSeededRandomPool prng_2;
-
-    ECIES<ECP>::PublicKey publicKey_1;
-    publicKey_1.Load(StringStore(addr_1_new.getEncryptionPublicKey()).Ref());
-    ECIES<ECP>::Encryptor encryptor_1(publicKey_1);
+    // TODO: when h_Sig is constructed properly as per spec
+    //       replace uint256() with it
+    ZCNoteEncryption encryptor = ZCNoteEncryption(uint256());
+    {
+        std::vector<unsigned char> plaintext_internals;
+        plaintext_internals.insert(plaintext_internals.end(), c_1_new.coinValue.begin(), c_1_new.coinValue.end());
+        plaintext_internals.insert(plaintext_internals.end(), c_1_new.r.begin(), c_1_new.r.end());
+        plaintext_internals.insert(plaintext_internals.end(), c_1_new.rho.begin(), c_1_new.rho.end());
 
-    std::vector<unsigned char> ciphertext_1_internals;
-    ciphertext_1_internals.insert(ciphertext_1_internals.end(), c_1_new.coinValue.begin(), c_1_new.coinValue.end());
-    ciphertext_1_internals.insert(ciphertext_1_internals.end(), c_1_new.r.begin(), c_1_new.r.end());
-    ciphertext_1_internals.insert(ciphertext_1_internals.end(), c_1_new.rho.begin(), c_1_new.rho.end());
+        std::vector<unsigned char> memo(ZC_MEMO_SIZE, 0x00);
+        plaintext_internals.insert(plaintext_internals.end(), memo.begin(), memo.end());
 
-    assert(ciphertext_1_internals.size() == (ZC_V_SIZE + ZC_R_SIZE + ZC_RHO_SIZE));
+        assert(plaintext_internals.size() == 216);
 
-    byte gEncryptBuf[encryptor_1.CiphertextLength(ZC_V_SIZE + ZC_R_SIZE + ZC_RHO_SIZE)];
+        boost::array<unsigned char, 216> pt;
+        memcpy(&pt[0], &plaintext_internals[0], 216);
 
-    encryptor_1.Encrypt(prng_1, &ciphertext_1_internals[0], ZC_V_SIZE + ZC_R_SIZE + ZC_RHO_SIZE, gEncryptBuf);
-
-    std::string C_1_string(gEncryptBuf, gEncryptBuf + sizeof gEncryptBuf / sizeof gEncryptBuf[0]);
-    this->ciphertext_1 = C_1_string;
-
-    ECIES<ECP>::PublicKey publicKey_2;
-    publicKey_2.Load(StringStore(addr_2_new.getEncryptionPublicKey()).Ref());
-    ECIES<ECP>::Encryptor encryptor_2(publicKey_2);
+        this->ciphertext_1 = encryptor.encrypt(addr_1_new.getEncryptionPublicKey(),
+                                               pt);
+    }
+    {
+        std::vector<unsigned char> plaintext_internals;
+        plaintext_internals.insert(plaintext_internals.end(), c_2_new.coinValue.begin(), c_2_new.coinValue.end());
+        plaintext_internals.insert(plaintext_internals.end(), c_2_new.r.begin(), c_2_new.r.end());
+        plaintext_internals.insert(plaintext_internals.end(), c_2_new.rho.begin(), c_2_new.rho.end());
```

Comment:
All code in `zerocash` is being deleted and I am only updating (minimally) here to avoid a large PR. In particular this code is deduplicated in the new PR.